### PR TITLE
Change default container user to root

### PR DESCRIPTION
**Pull request by alisonlhart**
This changes the default container user for both podman and docker containers to `root` instead of `root` and `podman` respectively. 
This is to resolve naming confusion as well as Ansible VSCode extension container-in-container issues. 
The volume mount removal is also due to container-in-container file copy problems that prevented syntax-highlighting and hover features. 

Related: https://issues.redhat.com/browse/AAP-30231

### CODE REVIEW

**Review by shatakshiiii — APPROVED** [review]

**Review by ganeshrn — APPROVED** [review]

**Review by ssbarnea — DISMISSED** [review]
That kind of change might have serious implications but I do not see any kind of change in the testing that would prevent us from undoing it in the future. Whatever we fix by it must be tested by at least on test.

File: final/Containerfile [hunk 1]
```diff
@@ -160,8 +160,6 @@ RUN set -ex \
 # Make a workdir usable by either the podman or root user
 RUN mkdir -p /workdir && chown podman:podman /workdir
 
-# Set the default user to podman, this will be overridden in the podman dev container
-USER podman
```
Comment (cidrblock) [inline final/Containerfile:164]: Can we also remove the configuration of the podman user from the containerfile since we're not going to use it any more?

**Review by cidrblock — CHANGES_REQUESTED** [review]

File: final/Containerfile (hunk 1 above)
Comment (alisonlhart) [reply final/Containerfile:164]: Definitely :+1:

Comment (alisonlhart) [reply final/Containerfile:164]: @cidrblock Can you re-review this portion?

Comment (cidrblock) [reply final/Containerfile:164]: Looks good

**Comment by alisonlhart** [comment]
I have added a workaround in this PR because the new test will not pass until the upstream image has been updated. This is using the `--user==root` param in the podman call. We can remove this later.

**Pull request merged**